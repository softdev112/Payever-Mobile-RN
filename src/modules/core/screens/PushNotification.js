import { Navigation } from 'react-native-navigation';
import { TouchableWithoutFeedback } from 'react-native';
import { Icon, StyleSheet, Text, View } from 'ui';

export default function PushNotification(
  { action, message, style, textStyle }: NotifPropTypes
) {
  function runNotificationAction() {
    Navigation.dismissInAppNotification();

    if (action) {
      action();
    }
  }

  return (
    <View style={[styles.container, style]}>
      <TouchableWithoutFeedback onPress={runNotificationAction}>
        <View style={styles.notifContent}>
          <Icon style={styles.icon} source="icon-payever-64" />
          <Text style={[styles.message, textStyle]}>{message}</Text>
        </View>
      </TouchableWithoutFeedback>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    height: 60,
    width: '100%',
    justifyContent: 'center',
    paddingHorizontal: 8,
    backgroundColor: 'yellowgreen',
  },

  notifContent: {
    flexDirection: 'row',
    alignItems: 'center',
  },

  icon: {
    fontSize: 16,
    color: '#000',
  },

  message: {
    marginLeft: 10,
    color: '#FFF',
    fontSize: 16,
  },
});

type NotifPropTypes = {
  action?: () => void;
  message?: string;
  style?: Object;
  textStyle?: Object;
};