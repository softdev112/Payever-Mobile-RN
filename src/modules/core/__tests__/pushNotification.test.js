/* eslint-disable max-len */
import React from 'react';
import renderer from 'react-test-renderer';
import PushNotification from '../screens/PushNotification';

jest.mock('../../../common/Navigation');

describe('modules/core/screens/PushNotification', () => {
  it('PushNotification should render correctly with dummy data', async () => {
    const action = jest.fn();

    const tree = renderer.create(
      <PushNotification
        action={action}
        message="Hello World!!!!!"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});