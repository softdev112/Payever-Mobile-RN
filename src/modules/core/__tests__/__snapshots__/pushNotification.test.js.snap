// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules/core/screens/PushNotification PushNotification should render correctly with dummy data 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "yellowgreen",
        "height": 60,
        "justifyContent": "center",
        "paddingHorizontal": 8,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      source="icon-payever-64"
      style={
        Array [
          Object {
            "fontFamily": "payeverIcons",
            "fontSize": 64,
          },
          Array [
            Object {
              "fontFamily": "payeverIcons",
              "fontSize": 64,
            },
            Object {
              "color": "#000",
              "fontSize": 16,
            },
          ],
        ]
      }
    >
      îªŸ
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#3d3d3d",
            "fontFamily": "Helvetica Neue",
            "fontWeight": "200",
          },
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 16,
              "marginLeft": 10,
            },
            undefined,
          ],
        ]
      }
    >
      Hello World!!!!!
    </Text>
  </View>
</View>
`;
